toast-tk-web-agent/src/main/java/io/toast/tk/agent/web/AgentServerImpl.java

```diff
@@ -61,13 +72,59 @@ public boolean register(String ApiKey) {
 		return false;
 	}
 
+	private void openAliveSocket(String apiKey) throws InterruptedException, ExecutionException {
+		final String socketUrl = (app.getConfig().getWebAppUrl() + "/api/agent/stream/"+apiKey)
+				.replace("https://","wss://")
+				.replace("http://", "ws://")
+				.replace("//", "/")
+				.replace(":/", "://");
+
+		wsSlientProvider.get().prepareGet(socketUrl).execute(new WebSocketUpgradeHandler.Builder().addWebSocketListener(
+				new WebSocketTextListener() {
+					@Override
+					public void onOpen(WebSocket webSocket) {
+					}
+
+					@Override
+					public void onClose(WebSocket webSocket) {
+
+					}
+
+					@Override
+					public void onError(Throwable throwable) {
+
+					}
+
+					@Override
+					public void onMessage(String s) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1186)